ec/ecresource/deploymentresource/flatteners.go

```diff
@@ -83,6 +83,12 @@ func modelToState(d *schema.ResourceData, res *models.DeploymentGetResponse) err
 				return err
 			}
 		}
+
+		if observability := flattenObservability(res.Settings); observability != nil {
```

Comment:
I think checking the length is slightly more solid

```suggestion
		if observability := flattenObservability(res.Settings); len(observability) > 0 {
```